### PR TITLE
Minimal demo in C

### PR DESCRIPTION
**Pull request by ringerc**
Add a demo in C and a README

Signed-off-by: Craig Ringer <craig@2ndquadrant.com>

### CODE REVIEW

File: src/cdemo/cdemo.c [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+		depth = strtol(argv[2], &endptr, 10);
+		if (endptr == argv[2])
+			usage_die();
+	}
+	else
+	{
+		/* Default to single level calls */
+		depth = 1;
+	}
+	fprintf(stderr, " at depth %d", depth);
```
Comment (isaachier) [inline src/cdemo/cdemo.c]: Needs `\n`.

File: src/cdemo/cdemo.c [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+	if (argc > 3)
+	{
```
Comment (isaachier) [inline src/cdemo/cdemo.c]: I think clang-format would change this. Please add your files to `scripts/clang-format.sh` and run it to format these.

**Comment by isaachier** [comment]
Closing until further notice.

**Comment by ringerc** [comment]
Thanks.

I've run out of time at the moment and can't pursue much more if I can't
get the C++ version buildable with fixes for non-hunter stuff merged. I'll
be trying to revisit tracing once I can find a client lib I can effectively
deploy, or find time to help on the C one.

**Pull request closed without merging**